### PR TITLE
Set up ESLint

### PR DESCRIPTION
**Pull request by nvie**
Following up on https://github.com/liveblocks/liveblocks/pull/106#discussion_r840219431. This sets up ESLint within the `@liveblocks/client` package. I've kept the config pretty minimal for now and stuck with the defaults as much as possible.

Fixed all the low-hanging fruit in terms of easy fixes. This will set us up to incrementally start enforcing best practices over time.

### Checklist 🏁

- [x] Set this up for `@liveblocks/client` (this PR)
- [x] Set this up for `@liveblocks/node`
- [x] Set this up for `@liveblocks/react`
- [x] Set this up for `@liveblocks/redux`
- [x] Set this up for `@liveblocks/zustand`

Will set up a GitHub action to automatically enforce these lint rules are expected at a later moment in a different PR.

### CODE REVIEW

File: packages/liveblocks-client/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  rules: {
+    // ----------------------------------------------------------------------
+    // NOTE: Only temporarily turned off!
+    // These checks are still GOOD IDEAS to re-enable later on, but for right
+    // now they're too noisy to enable.
+    // ----------------------------------------------------------------------
+    "@typescript-eslint/ban-ts-comment": "off",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-non-null-asserted-optional-chain": "off",
+    "@typescript-eslint/no-non-null-assertion": "off",
```
Comment (nvie) [inline packages/liveblocks-client/.eslintrc.js:16]: These are currently giving errors, but I think they're good hints of things we may want to come back to later and fix, so I'm temporarily disabling them here for now.

File: packages/liveblocks-client/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    // -------------------------------
+    // Custom syntax we want to forbid
+    // -------------------------------
+    "no-restricted-syntax": [
+      "error",
+      // {
+      //   selector: "PrivateIdentifier",
+      //   message:
+      //     "Avoid private identifiers to reduce bundle size. Instead of using `#foo`, prefer using `private _foo`.",
+      // },
```
Comment (nvie) [inline packages/liveblocks-client/.eslintrc.js:43]: cc @GuillaumeSalles after you merge https://github.com/liveblocks/liveblocks/pull/106, we can uncomment this to enforce that we avoid private fields/methods in the future automatically.

File: packages/liveblocks-client/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      // {
+      //   selector: "ForOfStatement",
+      //   message:
+      //     "Avoid for..of loops in libraries, because they generate unneeded Babel iterator runtime support code in the bundle",
+      // },
+      // {
+      //   selector: "ForInStatement",
+      //   message:
+      //     "for..in loops are never what you want. Loop over Object.keys() instead.",
+      // },
```
Comment (nvie) [inline packages/liveblocks-client/.eslintrc.js:53]: Perhaps these will be good to enforce, too at that point.

**Review by ofoucherot — APPROVED** [review]

**Comment by nvie** [comment]
I've added the ESLint configs to all the package directories now 👍  Will set up a GitHub action for this a bit later / in another PR.

File: packages/liveblocks-client/.eslintrc.js (hunk 2 above)
Comment (nvie) [reply packages/liveblocks-client/.eslintrc.js:43]: This was fixed by #135.

**Pull request merged**